.github/scripts/setup-devcontainer.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -e
+
+echo "Setting up ApplicationInsights-Java development environment..."
+
+# Set up Gradle with proper JVM options
+export GRADLE_OPTS="-Dorg.gradle.daemon=true -Dorg.gradle.parallel=true -Dorg.gradle.caching=true -XX:MaxMetaspaceSize=512m"
+
+# Pre-download Gradle wrapper and dependencies
+echo "Pre-downloading Gradle wrapper and dependencies..."
+./gradlew --version
+
+# Pre-compile buildSrc to avoid recompilation during regular builds
+echo "Pre-compiling buildSrc..."
+./gradlew :buildSrc:build --quiet || true
+
+# Download and cache dependencies for faster builds
+echo "Downloading and caching build dependencies..."
+./gradlew dependencies --quiet || true
```

Comment:
why not run

./gradlew build

instead

which will both download dependencies and populate the local build cache?